src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -1881,30 +1881,52 @@ private String init(boolean initFull) {
             if (persistentId != null) {
                 setIdByPersistentId();
             }
+            
             if (this.getId() != null) {
                 // Set Working Version and Dataset by Datasaet Id and Version
-                dataset = datasetService.findDeep(this.getId());
+                
+                // We are only performing these lookups to obtain the database id
+                // of the version that we are displaying, and then we will use it
+                // to perform a .findFind(versionId); see below. 
+                
+                dataset = datasetService.find(this.getId());
+                
                 if (dataset == null) {
                     logger.warning("No such dataset: "+dataset);
                     return permissionsWrapper.notFound();
                 }
                 //retrieveDatasetVersionResponse = datasetVersionService.retrieveDatasetVersionById(dataset.getId(), version);
                 retrieveDatasetVersionResponse = datasetVersionService.selectRequestedVersion(dataset.getVersions(), version);
+                if (retrieveDatasetVersionResponse == null) {
```

Comment:
is this necessary? i.e. we already found the dataset, can there be one without a version here? (is this some edge case I'm not immediately seeing?)

Reply:
I guess I see now that it was moved up from lower down. That said see my comment lower.

Reply:
We may have had a real data corruption case where a dataset failed to get deleted, and was left in the database without versions. I wasn't thinking about it honestly, just feel like it's a good idea to check for nulls. (but yes, it was a copy-and-paste from further below)